sdk/core/core/CMakeLists.txt

```diff
@@ -4,13 +4,32 @@
 cmake_minimum_required (VERSION 3.10)
 set(TARGET_NAME "az_core")
 
-project (${TARGET_NAME} LANGUAGES C)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt)
+    file (STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/version.txt VERSION_STRING)
+    message(STATUS "Read version from version.txt: " ${VERSION_STRING})
+    string(REGEX MATCHALL "([0-9]+).([0-9]+).([0.9]+)(.*)" PRE_RELEASE ${VERSION_STRING})
 
+    set(AZ_VERSION_MAJOR ${CMAKE_MATCH_1} )
+    set(AZ_VERSION_MINOR ${CMAKE_MATCH_2} )
+    set(AZ_VERSION_PATCH ${CMAKE_MATCH_3} )
+    if(CMAKE_MATCH_4)
+        string(SUBSTRING ${CMAKE_MATCH_4} 1 -1 PRERELEASE_SUBSTRING )
```

Comment:
Can you please add a comment explaining what the 1, -1 mean here.

Reply:
Its CMake SUBSTRING syntax.
https://cmake.org/cmake/help/v3.10/command/string.html?highlight=substring#substring

Reply:
I get that, I was more concerned about why begin is 1. Your comment helps make sense of it now :)

> Trim off the "-" from the start of pre-release portion